BUILD.md

```diff
@@ -0,0 +1,170 @@
+# Build FreeOrion
+
+
+## Hardware and OS Requirements
+
+FreeOrion will compile on most recent Windows, Mac OS X and Linux operating
+systems. Other operating systems have reported to work, but support is not
+actively maintained by the FreeOrion developers. FreeOrion is developed for x86
+compatible processor architectures, other architectures haven't been tested.
+
+The FreeOrion source code makes heavy use of templates and requires much memory
+to compile ; 4GB RAM or more is recommended. To build FreeOrion, expect up to
+45 minutes on a Intel Core i5 system.
+
+
+## Required Software Dependencies
+
+FreeOrion depends on the following software to build:
+
+  * [Visual Studio] - 2015 for Windows Desktop Update 3 ; Windows only
+  * [Xcode] - 6.4 or later ; Mac OS X only
+  * [CMake] - 3.4 (Mac OS X) ; 3.1 or later (Other non-Windows)
+  * A C++11 compliant compiler - Other Operating Systems
+    * [GNU GCC] - 5.0 or later
+    * [Clang] - 3.9 or later
+  * [Python] - 2 or later, but not 3 or later
```

Comment:
In boost 1.56, the docs say Python 2.2 or later. I suspect FreeOrion requires something more like 2.7, which is the version included with the SDKs / installers.

Reply:
Good point, 2.0 is probably wrong. @Dilvish-fo are you aware of any minimum requirements in terms of the Python API (except the mentioned minimum requirements of Boost Python?

Reply:
The AI scripts make use of python 2.7 features (e.g. dict comprehensions).

Reply:
Thanks for pointing out @Morlic-fo .